code-challenges.test.js

```diff
@@ -14,39 +14,150 @@
+// TypeError: str1.toString.toUpperCase is not a function
+
+test ('a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.', () => {
+	const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+
+	const letterA = "a"
+	// Expected output: 
+	const output1 = ["Mango", "Apricot", "Peach"]
+	const letterE = "e"
+	// Expected output: 
+	const output2 = ["Cherry", "Blueberry", "Peach"]
+
+	expect (letterFinder(fruitArray, letterA)).toEqual(output1)
+	expect (letterFinder(fruitArray, letterE)).toEqual(output2)
+
+})
 
-const letterA = "a"
-// Expected output: ["Mango", "Apricot", "Peach"]
-const letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
 
 // b) Create the function that makes the test pass.
+// A function that takes in an array of strings and a single letter
+const letterFinder = (array1, str1) => {
+	
+	// A function that takes in a string and a single letter
+	filterOut = (testee, character) => {
+
+		// Store upper and lower cases of letter
+		const charLowerCase = character.toLowerCase()
+		const charUpperCase = character.toUpperCase()
+
+		// If the string contains either upper or lower case of the letter, return it
+		if (testee.includes(charLowerCase) || testee.includes(charUpperCase)){
+			return testee
+		}
+		else return false
+	}
+	// Add a function as the filtering decider for upper/lower casing issue
+	return array1.filter((element) => filterOut(element, str1))
+}
+
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
+
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
-
+// Good error: ReferenceError: fullHouse is not defined
+
+// Other errors:
+// TypeError: fullHouse(...).toEqual is not a function
+
+test('A function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.', () => {
+	
+	const hand1 = [5, 5, 5, 3, 3]
+	// Expected output: true
+	const hand2 = [5, 5, 3, 3, 4]
+	// Expected output: false
+	const hand3 = [5, 5, 5, 5, 4]
+	// Expected output: false
+	const hand4 = [7, 2, 7, 2, 7]
+	// Expected output: true
+
+	expect (fullHouse(hand1)).toEqual(true)
+	expect (fullHouse(hand2)).toEqual(false)
+	expect (fullHouse(hand3)).toEqual(false)
+	expect (fullHouse(hand4)).toEqual(true)
+})
 // b) Create the function that makes the test pass.
+// A function, that takes in an array of 5 numbers
+const fullHouse = (array1) => {
```

Comment:
It seems like you are using nested iteration for this problem, which may not be necessary.  I like the idea of initializing a variable to count the amount of times you run across the same number, but keep in mind you will want to keep track of more than just one instance if it is a full house.